src/pygeosx/PyGroup.cpp

```diff
@@ -250,14 +254,31 @@ static PyObject * PyGroup_getWrapper( PyGroup * const self, PyObject * const arg
     return nullptr;
   }
 
-  string groupPath, wrapperName;
-  std::tie( groupPath, wrapperName ) = splitPath( path );
-
   try
   {
-    dataRepository::Group & group = self->group->getGroupByPath( groupPath );
-    dataRepository::WrapperBase & wrapper = group.getWrapperBase( wrapperName );
-    return createNewPyWrapper( wrapper );
+    geosx::ProblemManager * pbManager = static_cast< geosx::ProblemManager * >(self->group);
+    geosx::EventManager & eventManager = pbManager->getEventManager();
+    eventManager.forSubGroups< geosx::EventBase >( [&]( geosx::EventBase & subEvent )
+    {
+      subEvent.getTargetReferences();
+    } );
+
+    geosx::EventBase * subEvent = nullptr;
+    for( int currentSubEvent = 0; currentSubEvent<eventManager.numSubGroups(); ++currentSubEvent )
+    {
+      subEvent = static_cast< geosx::EventBase * >( eventManager.getSubGroups()[currentSubEvent]);
+      if( subEvent->getEventName() == path )
```

Comment:
So here you really just pass the name of the solver right, and not the full path? And then if you input something that is an `Event` but not a `Solver` what happens? Is there not a better way to loop over the physics solvers than to loop over all events?

Reply:
That is right, the user just gives the name of the Solver not the full path. 
Yes you're right this can be an issue. I don't know if there is a way to loop only over the Solver I will check that. Could we also imagine checking the dynamic type of the event ?

Reply:
I'd like to see how you get one of these solver objects from Python. My understanding is that it's something like this

``` python
problem = geosx.initialize(...)
solver = geosx.PySolver.solver('solverName')
```

My suggestion is that it is done as follows

``` python
problem = geosx.initialize(...)
solver = geosx.getGroup('path/to/solvers/solverName')
```

Then internally the C++ function returns a `geosx.PySolver.solver` type (which inherits from `geosx.Group`) instead of the `geosx.Group` object. This could easily be extended to support whatever arbitrary group types we want to export to python (including the time history stuff). Yes this requires that you specify the full path to the solver, but when combined with @cssherman's existing pure-python `pygeosx` helpers this is not a problem.

Reply:
You're right this is the way it is done on `Python` side.
I see what you mean and this is kind of what I did in the second version of `PySolver` `(PySolver2)` except that I had a different routine `getSolver` which creates the `PySolver` object. What you propose is that `getGroup` does actually all sort of `PyStructure` creation. In my understanding this would mean that we would have some sort of mechanism to know which `PyStructure` to build based on the `path` of the `group` right ?

Reply:
You could add a function
```c++
PyTypeObject * getPythonType() virtual const
{ return getPyGroupType(); }
```

to `Group` and then derived types could override it with their own type (should inherit from the Python `Group` class), then all you'd have to do is in `createNewPyGroup` replace `getPyGroupType()` with `group.getPythonType()`. It'd take a bit of trickery to avoid circular references, but I think something like that would be best. If we're going to have more of these python types we'll be exporting perhaps we should move them out of the `pygeosx` directory and put them under `dataRepository/python` for example.

Reply:
Hi Ben, first I wish you a happy new year and hope you had good holidays ! 
I just pushed a commit that integrates this modification for creation of the different `PyStructure` (including the `PyHistory` one that I broke in 2 different structures being part of the new `PyHistory` `submodule` of `pygeosx` `module)`. We could easily imagine pure `Python` object taking those `PyStructures` to make it easier for users, and as you said, combined with @cssherman existing pure-python, makes it even more easy. 
Please let me know what you think of that new version. You also have a dummy test file in the `pygeosx` directory with `xml` associated with it if you want to see how it works.

Reply:
Thanks, you too!

I'm bit busy at the moment, but I'm looking forward to checking it out when I have some free time.